src/Bicep.Core/Semantics/Namespaces/SystemNamespaceType.cs

```diff
@@ -113,6 +113,38 @@ maxLength switch
                 .WithDescription("Combines multiple arrays and returns the concatenated array.")
                 .WithVariableParameter("arg", LanguageConstants.Array, minimumCount: 1, "The array for concatenation")
                 .Build();
+            
+            yield return new FunctionOverloadBuilder("toLogicalZone")
+                .WithReturnType(LanguageConstants.String)
```

Comment:
Can the result of this function be known at compile time (like `toLower('FOO')`), or is this something that can only be decided at deploy time (like `resourceGroup()`)?

If this function is decidable at compile time, you can use `.WithReturnResultBuilder(TryDeriveLiteralReturnType("toLogicalZone", LanguageConstants.String), LanguageConstants.String)` instead of `.WithReturnType(LanguageConstants.String)` to have the type system try to execute this function on its arguments and derive a literal return type.

If this function is only decidable at deploy time (e.g., if the mapping is based on ARM's service configuration or varies from subscription to subscription), it should probably be defined in `AzNamespaceType` (like `resourceGroup()` is), which would make it part of the `az` namespace.

Reply:
Good point! The fact that it contains a parameter named `subscriptionId` tells me it probably belongs in `az` regardless.